### PR TITLE
Add abilility to use traffic distribution on old k8s clusters

### PR DESCRIPTION
**Pull request by howardjohn**
TrafficDistribution is alpha in k8s 1.30. So it will be 2-3 years before
its something we can depend on.

We really want to make waypoints PreferClose, so we hardcoded a waypoint
hack. But this leaves all other services out of the picture.

This adds a temporary (where temporary is ~2-3 years) annotation to
allow users on Kubernetes clusters w/o this field to utilize ztunnel
locality aware load balancing. Additionally, the waypoint hack is
dropped and moved to use this annotation. Conveniently, this also means
a user can chose to opt-out of this, if they want.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml
Comment (zirain) [inline manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:298]: I used to think that all annotations is experimental before promote to offical API.

File: pilot/pkg/serviceregistry/kube/controller/ambient/services.go
Comment (ilrudie) [inline pilot/pkg/serviceregistry/kube/controller/ambient/services.go:191]: Should "PreferClose" be a constant? I think we could also use `v1.ServiceTrafficDistributionPreferClose`

Comment (ilrudie) [reply pilot/pkg/serviceregistry/kube/controller/ambient/services.go:191]: Also I guess worth considering, do we want this to be case sensitive?

**Review by ilrudie — APPROVED** [review]
lgtm

**Review by zirain — APPROVED** [review]

File: pilot/pkg/serviceregistry/kube/controller/ambient/services.go
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/ambient/services.go:151]: should this be in the api repo

File: pilot/pkg/serviceregistry/kube/controller/ambient/services.go
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/ambient/services.go:246]: nit: define a var

**Review by keithmattix — APPROVED** [review]

File: pilot/pkg/serviceregistry/kube/controller/ambient/services.go
Comment (howardjohn) [reply pilot/pkg/serviceregistry/kube/controller/ambient/services.go:151]: yeah I think we can probably move it over once this is merged? No point getting the API approved if we don't approve of adding it at all

**Review by hzxuzhonghu — APPROVED** [review]
Can replace with annotation from api later

**Pull request merged**
